crates/noirc_frontend/src/hir_def/types.rs

```diff
@@ -222,16 +241,31 @@ impl TypeBinding {
     }
 }
 
+/// A unique ID used to differentiate different type variables
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
 pub struct TypeVariableId(pub usize);
 
+/// Noir's type system keeps track of whether or not every primitive type's value
+/// is known at compile-time. This is exposed through users through the `comptime`
+/// keyword in noir which can be prefixed before primitive types. A usage like
+/// `t: comptime Field` would correspond to a Field type with a CompTime::Yes(None)
+/// variant of this enum
+///
+/// Note that whether or not a variable is comptime can also be inferred based on its use.
+/// A value passed to a function that expects a `comptime Field` must be CompTime::Yes,
+/// likewise a parameter of the current function that is just a `Field` can only be CompTime::No.
+/// There is also the case of integer literals which are typed as CompTime::Maybe. These are
+/// polymorphically comptime because they can be used in both contexts.
 #[derive(Debug, Clone, Eq)]
 pub enum CompTime {
     // Yes and No variants have optional spans representing the location in the source code
     // which caused them to be compile time.
     Yes(Option<Span>),
     No(Option<Span>),
-    Maybe(TypeVariableId, Rc<RefCell<Option<CompTime>>>),
+
+    /// Maybe has an id and shared inner rebindable reference. Similar to type variables rebinding
```

Comment:
rebindable does not seem to be a word, can you rephrase?

Reply:
Seems like another rare term the spell checker is unaware of (e.g. it is also used by haskell: https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/rebindable_syntax.html). "shared inner reference that can be rebound" perhaps.